README.md

```diff
@@ -1,65 +1,215 @@
+```
+
+Check the number of rows in the demo.tt10000 table and get the sum for column c
+
+```
+tidiff 'select count(*) as count, sum(c) from demo.tt10000'
+```
+
+Select 5 random rows from the demo.tt10000 table
+
+```
+tidiff 'select * from demo.tt10000 order by rand() limit 5'
+```
+
+Here is the screenshot of the operation above. The result differences are displayed in diff format.
 
-    You can use the command line mode as downstream pipeline, e.g: `randgen | xargs tididff`. The SQL statement should be quote with `'` instead of `"`.
+![](media/tidiff-guide-demo1.png)
 
-- UI Mode Shortcuts
+You can use the command line mode as downstream pipeline, e.g: `randgen | xargs tididff`. The SQL statement should be quote with "` instead of `".
+
+## Interactive Mode
+
+tidiff provides an interactive mode which records SQL statements execution history so as to run a SQL statement repeatedly. 
+
+- To enable the interactive mode, enter `tidiff --mysql.db test --tidb.db test` without a SQL statement.
+
+- If the interactive mode includes `--log.diff` parameters like tidiff `--log.diff /path/to/diff.log`, then all the results will be saved into the log file. The difference will also be highlighted when you view the results using the `cat` command.  
+
+- If a SQL statement that includes some Golang template fails, the error part will be highlighted with `/* error message */` at the end of the statement.  
+
+    ![DEMO](media/tidiff-guide-ui.png)
+
+- Shortcuts for interactive mode
+
+    - SQL Input Panel
+
+        - Use `Enter` to execute SQL statements. If a SQL statement begins with “!” , then Golang template is firstly used, where you can embed statements or expressions that generate random data. If a SQL statement begins with “!!”, the generated SQL statement will appear on the output panel. The SQL statement rendered will not be output in the output panel by default. 
+
+        - Use `TAB` to switch between panels. 
+
+        - Use `Up/Dn` to fast shift the focus to the History panel.
+
+    - MySQL/TiDB Output Panel 
+
+        - Use `Up/Dn` to turn page up or down.
+
+        - Use `TAB` to switch between panels.
     
-    - `TAB` switch focus in different panel
-    - `ESC` focus on SQL input panel
-    - `UP`  focus on the previous history item
-    - `DOWN` focus on the next history item
-    - `ENTER` select current selected history item
+        - Use `ESC` and return to the SQL input panel. 
+
+    - History Panel 
+
+        - Use `Up/Dn` to fast shift the focus to the SQL input panel. 
     
-    ![DEMO](media/tidiff-guide-ui.png)
+        - Select a history entry and use `Enter` to fill it in the SQL input panel for later editing and executing.
+
+        - Use `ESC` and return to the SQL input panel.
+ 
+## Golang template
 
-## Features
+Here only introduces usages of “:=”, “range”, and “if”, and the functions provided in tidiff. All SQL statements must start with “!" so that tidiff can render them using the Golang template. 
```

Comment:
Please refer to [the original Chinese file](https://github.com/lonng/notes/blob/master/blog/the-tidiff-guide.md) and add the inline code format for necessary characters in this PR, not only in this line.

Reply:
> Please refer to [the original Chinese file](https://github.com/lonng/notes/blob/master/blog/the-tidiff-guide.md) and add the inline code format for necessary characters in this PR, not only in this line.

@huangxiuyan Could you please be more careful when you read comments? I'd appreciate it a lot because I don't want to remind you of necessary updates over and over again.